### PR TITLE
fix: add ipv6 nic config for vmss [IPv6DualStack]

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Add ipv6 nic config to vmss for dualstack support
- Rename default ipv4 default backend pool name to match cluster name to support backward compat
- Add `ip6_tables` module

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/loadbalancers.go [hunk 1]
```diff
@@ -31,7 +31,8 @@ func CreateClusterLoadBalancerForIPv6() LoadBalancerARM {
 			LoadBalancerPropertiesFormat: &network.LoadBalancerPropertiesFormat{
 				BackendAddressPools: &[]network.BackendAddressPool{
 					{
-						Name: to.StringPtr("[concat(parameters('masterEndpointDNSNamePrefix'), '-ipv4')]"),
+						// cluster name used as backend addr pool name for ipv4 to ensure backward compat
+						Name: to.StringPtr("[concat(parameters('masterEndpointDNSNamePrefix'))]"),
```
[the remark this answers is not in the document]
Comment (aramase) [reply pkg/engine/loadbalancers.go]: Done

File: pkg/engine/loadbalancers.go [hunk 2]
```diff
@@ -77,7 +78,7 @@ func CreateClusterLoadBalancerForIPv6() LoadBalancerARM {
 								ID: to.StringPtr("[resourceId('Microsoft.Network/loadBalancers/frontendIpConfigurations', parameters('masterEndpointDNSNamePrefix'), 'LBFE-v4')]"),
 							},
 							BackendAddressPool: &network.SubResource{
-								ID: to.StringPtr("[resourceId('Microsoft.Network/loadBalancers/backendAddressPools', parameters('masterEndpointDNSNamePrefix'), concat(parameters('masterEndpointDNSNamePrefix'), '-ipv4'))]"),
+								ID: to.StringPtr("[resourceId('Microsoft.Network/loadBalancers/backendAddressPools', parameters('masterEndpointDNSNamePrefix'), concat(parameters('masterEndpointDNSNamePrefix')))]"),
```
[the remark this answers is not in the document]
Comment (aramase) [reply pkg/engine/loadbalancers.go]: Done

File: pkg/engine/loadbalancers_test.go [hunk 3]
```diff
@@ -580,7 +580,7 @@ func TestCreateClusterLoadBalancerForIPv6(t *testing.T) {
 			LoadBalancerPropertiesFormat: &network.LoadBalancerPropertiesFormat{
 				BackendAddressPools: &[]network.BackendAddressPool{
 					{
-						Name: to.StringPtr("[concat(parameters('masterEndpointDNSNamePrefix'), '-ipv4')]"),
+						Name: to.StringPtr("[concat(parameters('masterEndpointDNSNamePrefix'))]"),
```
[the remark this answers is not in the document]
Comment (aramase) [reply pkg/engine/loadbalancers_test.go]: Done

File: pkg/engine/loadbalancers_test.go [hunk 4]
```diff
@@ -626,7 +626,7 @@ func TestCreateClusterLoadBalancerForIPv6(t *testing.T) {
 								ID: to.StringPtr("[resourceId('Microsoft.Network/loadBalancers/frontendIpConfigurations', parameters('masterEndpointDNSNamePrefix'), 'LBFE-v4')]"),
 							},
 							BackendAddressPool: &network.SubResource{
-								ID: to.StringPtr("[resourceId('Microsoft.Network/loadBalancers/backendAddressPools', parameters('masterEndpointDNSNamePrefix'), concat(parameters('masterEndpointDNSNamePrefix'), '-ipv4'))]"),
+								ID: to.StringPtr("[resourceId('Microsoft.Network/loadBalancers/backendAddressPools', parameters('masterEndpointDNSNamePrefix'), concat(parameters('masterEndpointDNSNamePrefix')))]"),
```
[the remark this answers is not in the document]
Comment (aramase) [reply pkg/engine/loadbalancers_test.go]: Done

File: parts/k8s/cloud-init/artifacts/cse_main.sh [hunk 5]
```diff
@@ -160,6 +160,8 @@ if [ "$IS_IPV6_DUALSTACK_FEATURE_ENABLED" = "true" ]; then
     wait_for_file 3600 1 $dhcpv6_systemd_service || exit $ERR_FILE_WATCH_TIMEOUT
     wait_for_file 3600 1 $dhcpv6_configuration_script || exit $ERR_FILE_WATCH_TIMEOUT
     ensureDHCPv6
+
+    retrycmd_if_failure 120 5 25 modprobe ip6_tables || exit $ERR_MODPROBE_FAIL
```
Comment (CecileRobertMichon) [inline parts/k8s/cloud-init/artifacts/cse_main.sh]: not a blocker for this PR but as a follow up could you refactor the ipv6 cse setup into a function in cse_config.sh ?

**Pull request merged**